fundrebalancer/fundrebalancer.go

```diff
@@ -244,21 +244,45 @@ func (r *FundRebalancer) MoveFundsToChain(
 			return nil, nil, fmt.Errorf("getting txns required for fund rebalancing: %w", err)
 		}
 
+		chainConfig, err := config.GetConfigReader(ctx).GetChainConfig(rebalanceFromChainID)
+		if err != nil {
+			return nil, nil, fmt.Errorf("getting chain config for gas threshold check: %w", err)
+		}
+
+		if chainConfig.MaxRebalancingGasThreshold != 0 {
+			// Check if total gas needed exceeds threshold to rebalance funds from this chain
+			totalRebalancingGas, err := r.estimateTotalGas(txns)
+			if err != nil {
+				return nil, nil, fmt.Errorf("estimating total gas for transactions: %w", err)
+			}
+
+			if totalRebalancingGas > chainConfig.MaxRebalancingGasThreshold {
+				lmt.Logger(ctx).Info(
+					"skipping rebalance from chain "+rebalanceFromChainID+" due to rebalancing txs exceeding gas threshold",
+					zap.String("sourceChainID", rebalanceFromChainID),
+					zap.String("destinationChainID", rebalanceToChain),
+					zap.Uint64("estimatedGas", totalRebalancingGas),
+					zap.Uint64("gasThreshold", chainConfig.MaxRebalancingGasThreshold),
+				)
+				continue
+			}
+		}
+
 		signedTxns, err := r.SignTxns(ctx, txns)
 		if err != nil {
 			return nil, nil, fmt.Errorf("signing txns required for fund rebalancing: %w", err)
 		}
 
 		txnHashes, err := r.SubmitTxns(ctx, signedTxns)
 		if err != nil {
-			return nil, nil, fmt.Errorf("submitting signed txns required for fund rebalancing: %w", err)
+			return nil, nil, fmt.Errorf("error submitting signed txns required for fund rebalancing: %w", err)
```

Comment:
a personal nit of mine but remove the "error..." from the start of the wrap. its more convention to say just what happened in the wrapped error message, instead of saying "error..." or "failed..." at the start